### PR TITLE
Localized rescue for crashy button clicks

### PR DESCRIPTION
**Pull request by twometresteve**
## Goal

Restrict the rescuing of errors on Appium button clicks to just those that occur for `run_scenario` (and on macOS).

## Design

Until now such errors were handled by Maze Runner for all button clicks, but this will be removed in [its next release](https://github.com/bugsnag/maze-runner/pull/285) as it is too generic.

## Testing

Covered by CI.

### CODE REVIEW

**Comment by nickdowell** [comment]
💭 In theory only the `When("I run {string} and relaunch the app")` step should ever trigger crashes - so we might be able to restrict the rescue to that step and be more specific

**Comment by twometresteve** [comment]
> 💭 In theory only the `When("I run {string} and relaunch the app")` step should ever trigger crashes - so we might be able to restrict the rescue to that step and be more specific

Yes, agreed - now reworked.

**Review by nickdowell — APPROVED** [review]

File: features/steps/ios_steps.rb [hunk 1]
```diff
@@ -16,10 +16,15 @@
...
 
 When("I run {string} and relaunch the app") do |event_type|
-  steps %Q{
-    When I run "#{event_type}"
-    And I relaunch the app after a crash
-  }
+  begin
+    step("I run \"#{event_type}\"")
+  rescue StandardError
+    # AppiumForMac raises an to run a scenario that crashes the app
```
Comment (nickdowell) [inline features/steps/ios_steps.rb:22]: I'm finding it difficult to understand this comment 😅

Comment (twometresteve) [reply features/steps/ios_steps.rb]: Seems picky 😂 , but reworded.

**Pull request merged**